internal/baggage/remote/restriction_manager.go

```diff
@@ -72,16 +78,12 @@ func NewRestrictionManager(serviceName string, options ...Option) *RestrictionMa
 		serviceName:        serviceName,
 		options:            opts,
 		restrictions:       make(map[string]*baggage.Restriction),
-		thriftProxy:        &httpBaggageRestrictionManagerProxy{serverURL: opts.serverURL},
+		thriftProxy:        newHTTPBaggageRestrictionManagerProxy(opts.hostPort, serviceName),
 		stopPoll:           make(chan struct{}),
 		invalidRestriction: baggage.NewRestriction(false, 0),
 		validRestriction:   baggage.NewRestriction(true, defaultMaxValueLength),
 	}
-	if err := m.updateRestrictions(); err != nil {
-		m.logger.Error(fmt.Sprintf("Failed to initialize baggage restrictions: %s", err.Error()))
-	} else {
-		m.initialized = true
-	}
+	go m.updateRestrictions() // initialize restrictions asynchronously
```

Comment:
the problem with this is that if you get an error it won't be logged. Maybe you can fix pollManager() function to do the first update right away, without waiting for the ticker?